tests/test_dns.py

```diff
@@ -31,13 +31,13 @@ def _test_getaddrinfo(self, *args, _patch=False, **kwargs):
                 a1 = patched_getaddrinfo(*args, **kwargs)
             else:
                 a1 = socket.getaddrinfo(*args, **kwargs)
-        except socket.gaierror as ex:
+        except (socket.gaierror, UnicodeError) as ex:
             err = ex
 
         try:
             a2 = self.loop.run_until_complete(
                 self.loop.getaddrinfo(*args, **kwargs))
-        except socket.gaierror as ex:
+        except (socket.gaierror, UnicodeError) as ex:
```

Comment:
What input would trigger a UnicodeError?

Reply:
This is what @tapple-cisco mentioned with the vulnerability repro, as well as the [BPO](https://github.com/python/cpython/issues/77139). A short example with CPython is like:

```python
>>> payload = f'0x{"0"*246}7f000001.example.com'
>>> import socket; socket.getaddrinfo(payload, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.12/socket.py", line 964, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/encodings/idna.py", line 173, in encode
    raise UnicodeError("label empty or too long")
UnicodeError: label empty or too long
encoding with 'idna' codec failed
```

Reply:
Is `encoding with 'idna' codec failed` a context exception?

Reply:
i.e. `.__context__`.

Reply:
yeah it's a weird output, let me see

Reply:
It's not context, but a `_PyErr_FormatNote` [added in 3.12](https://github.com/python/cpython/pull/103402)